.circleci/workflows.yml

```diff
@@ -328,7 +343,80 @@ jobs:
           name: Download dist artifacts and run the self-test tool
           command: ferrocene/ci/scripts/run-self-test.sh
 
-  # aarch64-apple-darwin jobs
+  # aarch64-linux jobs
+
+  aarch64-linux-llvm:
+    executor: docker-aarch64-ubuntu-20
+    resource_class: arm.xlarge # 8-core
+    environment:
+      FERROCENE_HOST: aarch64-unknown-linux-gnu
+      SCRIPT: ferrocene/ci/scripts/llvm-cache.sh prepare
+      # This tries to strike a balance between the highest concurrency possible
+      # and not running out of memory during a build. If building LLVM fails
+      # due to the OOM killer killing the build, decrease this number.
+      LLVM_BUILD_PARALLELISM: 18
+    steps:
+      - when:
+          condition: << pipeline.parameters.llvm-rebuild--aarch64-unknown-linux-gnu >>
+          steps:
+            - aws-oidc-auth
+            - ferrocene-checkout
+            - ferrocene-ci
+      - run:
+          name: Empty step to make sure the job always has steps
+          command: echo
+
+  aarch64-linux-build:
+    executor: docker-aarch64-ubuntu-20
+    resource_class: arm.xlarge # 8-core
+    environment:
+      FERROCENE_HOST: aarch64-unknown-linux-gnu
+      FERROCENE_TARGETS: << pipeline.parameters.targets--aarch64-unknown-linux-gnu--build >>
+      SCRIPT: |
+        ferrocene/ci/scripts/llvm-cache.sh download
+        ./x.py --stage 2 build library src/tools/rustdoc
+        ./x.py --stage 1 build src/tools/rustdoc
```

Comment:
Not PR specific but: Why are we first building stage 2 and then stage 1?

Reply:
Because some tests need stage 1 rustdoc and some other tests need stage 2 rustdoc. We thus build both of them. The order doesn't really matter.